cdc/cdc.cc

```diff
@@ -64,12 +65,119 @@ using namespace std::chrono_literals;
 
 static logging::logger cdc_log("cdc");
 
-class cdc::cdc_service::impl {
+namespace cdc {
+static schema_ptr create_log_schema(const schema&, std::optional<utils::UUID> = {});
+static schema_ptr create_stream_description_table_schema(const schema&, std::optional<utils::UUID> = {});
+static future<> populate_desc(db_context ctx, const schema& s);
+}
+
+class cdc::cdc_service::impl : service::migration_listener::empty_listener {
     db_context _ctxt;
 public:
     impl(db_context ctxt)
         : _ctxt(std::move(ctxt))
-    {}
+    {
+        if (engine().cpu_id() == 0) {
+            _ctxt._migration_manager.register_listener(this);
+        }
+    }
+    ~impl() {
+        if (engine().cpu_id() == 0) {
+            _ctxt._migration_manager.unregister_listener(this);
+        }
+    }
+
+    void on_before_create_column_family(const schema& schema, std::vector<mutation>& mutations, api::timestamp_type timestamp) override {
+        if (schema.cdc_options().enabled()) {
+            auto& db = _ctxt._proxy.get_db().local();
+            auto logname = log_name(schema.cf_name());
+            if (!db.has_schema(schema.ks_name(), logname)) {
+                // in seastar thread
```

Comment:
While this comment is true, why is it relevant? Unless I'm missing something, all of the code below is immediate, and doesn't use get() or anything?

Reply:
Irrelevant, but useful in case anyone ever needs to introduce a continuation.